### DISCUSSION TITLE
How do you run qbitmanage via qB's pre-/post-run external program commands for qBittorrent in docker?

### DISCUSSION THREAD

**Discussion by glottisfaun0000** [StuffAnThings/qbit_manage#614] [Q&A] [answered]
I know how to exec into the container to install requirements but not how to do it so you don't lose the qbitmanage install whenever the container is recreated. Would you include the requirements install as a custom entrypoint? Or is there a way of putting like a portable install in a directory the container accesses so it's always available? I'm not that experienced with python obviously. Or do you run a parallel qbitmanage container and allow qB to trigger the other container somehow?

Plz share your configuration

**Comment by davidgibbons** [comment, marked as answer]
I was struggling with this and ended up building a child container that included a webhook: https://github.com/davidgibbons/qbit_webhook

This runs in its own container and then qBitTorrent calls a small curl script to trigger it on torrent start.

**Reply by glottisfaun0000** [reply]
sounds interesting - personally i would need an image, example compose, and example curl or whatever for this to be approachable but hope it helps someone else

**Reply by davidgibbons** [reply]
Thanks for the feedback, I updated the readme with those pieces.

**Reply by glottisfaun0000** [reply]
where do I place the hook.json? I put it in /config from the qbitmanage container's perspective and just get a 404 when curling. also should there be a hooks.yml?

**Reply by davidgibbons** [reply]
Oh yeah, it should totally be hooks.yml and not hook.json based on the default arguments, thats the right place to put it. I'll update the readme, thanks!
```
- id: qbitmanage
  execute-command: /app/qbit_manage.py
  command-working-directory: /app
```

**Reply by glottisfaun0000** [reply]
sweet, works for me now. just to make it clear for anyone else who comes across this, the block starting with `- id : qbitmanage` should be the contents of a file called hooks.yml in /config as defined for the qbit_webhook container. thanks a lot for sharing this btw, i see no reason this couldn't be wrapped into the main qbit_manage branch!

**Reply by glottisfaun0000** [reply]
one more thing, does this/could this support the :develop branch of qbit_manage? the image i use is `ghcr.io/stuffanthings/qbit_manage:develop` so that i have support for qbittorrent 5.0.1

**Reply by davidgibbons** [reply]
[That should be what it's based on](https://github.com/davidgibbons/qbit_webhook/blob/8c1a608923e265781a812551cf1c93a6d5b79a7d/Dockerfile#L11). I'm using 5.0.1 as well. 
Is it throwing an error that 5.0.1 isn't supported it?

**Reply by glottisfaun0000** [reply]
just came back to edit and mention it's actually working not throwing an error and I see `Version: 4.1.14-develop1` in the output so it works fine and is using :develop. no idea if that's worth adding as an option, thanks 👍

**Comment by pgm95** [comment]
If anyone is still looking, I posted a simple workaround in #779
